lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -69,6 +70,85 @@ export class PublicClientApplication extends ClientApplication implements IPubli
         return this.acquireTokenPopup(request || DEFAULT_REQUEST);
     }
 
+    /**
+     * This function uses a hidden iframe to fetch an authorization code from the eSTS. There are cases where this may not work:
+     * - Any browser using a form of Intelligent Tracking Prevention
+     * - If there is not an established session with the service
+     *
+     * In these cases, the request must be done inside a popup or full frame redirect.
+     *
+     * For the cases where interaction is required, you cannot send a request with prompt=none.
+     *
+     * If your refresh token has expired, you can use this function to fetch a new set of tokens silently as long as
+     * you session on the server still exists.
+     * @param {@link AuthorizationUrlRequest}
+     *
+     * @returns {Promise.<AuthenticationResult>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+     */
+    async ssoSilent(request: SsoSilentRequest): Promise<AuthenticationResult> {
+        this.preflightBrowserEnvironmentCheck(InteractionType.Silent);
+        this.emitEvent(EventType.SSO_SILENT_START, InteractionType.Silent, request);
+
+        try {
+            // Build cached account by loginHint, sid, or the request itself
+            const accountByLoginHint = request.loginHint && this.getAccountByUsername(request.loginHint);
+            
+            const accountBySid = request.sid && this.getAccountBySessionId(request.sid);
+            if (request.sid && !accountBySid) {
+                this.logger.warning("ssoSilent - sid provided to ssoSilent, but no cached account found. If this is unexpected, confirm sid is enabled as an optional id token claim for your application");
+            }
+
+            const account = request.account || accountByLoginHint || accountBySid;
```

Comment:
imo `sid` should come before `loginHint`
```suggestion
            const account = request.account || accountBySid || accountByLoginHint;
```

Reply:
Agreed.

Reply:
Just FYI rule of thumb: `user passed account` > `cached account` and `sid` > `loginHint` in any account read.